catalogs/catalog-hive/src/test/java/com/datastrato/gravitino/catalog/hive/TestHiveTableOperations.java

```diff
@@ -156,4 +156,96 @@ public void testAddPartition() {
             .contains("Hive partition field name must be in table partitioning field names"),
         exception.getMessage());
   }
+
+  @Test
+  public void testDropPartition() {
+    // add partition: city=2/dt=2020-01-01
+    String partitionName1 = "city=2/dt=2020-01-01";
+    String[] fieldCity1 = new String[] {columns[1].name()};
+    Literal<?> valueCity1 = Literals.byteLiteral((byte) 2);
+    String[] fieldDt1 = new String[] {columns[2].name()};
+    Literal<?> valueDt1 = Literals.dateLiteral(LocalDate.parse("2020-01-01"));
+    Partition partition1 =
+        Partitions.identity(
+            new String[][] {fieldCity1, fieldDt1}, new Literal<?>[] {valueCity1, valueDt1});
+    hiveTable.supportPartitions().addPartition(partition1);
+
+    // add partition: city=3/dt=2020-01-01
+    String partitionName2 = "city=3/dt=2020-01-01";
+    String[] fieldCity2 = new String[] {columns[1].name()};
+    Literal<?> valueCity2 = Literals.byteLiteral((byte) 3);
+    String[] fieldDt2 = new String[] {columns[2].name()};
+    Literal<?> valueDt2 = Literals.dateLiteral(LocalDate.parse("2020-01-01"));
+    Partition partition2 =
+        Partitions.identity(
+            new String[][] {fieldCity2, fieldDt2}, new Literal<?>[] {valueCity2, valueDt2});
+    hiveTable.supportPartitions().addPartition(partition2);
+
+    // add partition: city=3/dt=2020-01-02
+    String partitionName3 = "city=3/dt=2020-01-02";
+    String[] fieldCity3 = new String[] {columns[1].name()};
+    Literal<?> valueCity3 = Literals.byteLiteral((byte) 3);
+    String[] fieldDt3 = new String[] {columns[2].name()};
+    Literal<?> valueDt3 = Literals.dateLiteral(LocalDate.parse("2020-01-02"));
+    Partition partition3 =
+        Partitions.identity(
+            new String[][] {fieldCity3, fieldDt3}, new Literal<?>[] {valueCity3, valueDt3});
+    hiveTable.supportPartitions().addPartition(partition3);
+
+    // test drop one partition: city=2/dt=2020-01-01
+    hiveTable.supportPartitions().dropPartition(partitionName1, false);
+    NoSuchPartitionException exception1 =
+        Assertions.assertThrows(
+            NoSuchPartitionException.class,
+            () -> {
+              hiveTable.supportPartitions().getPartition(partitionName1);
+            });
+    Assertions.assertEquals(
+        String.format("Hive partition %s does not exist in Hive Metastore", partitionName1),
+        exception1.getMessage());
+
+    // test drop cascade partitions: city=3
+    hiveTable.supportPartitions().dropPartition("city=3", false);
```

Comment:
I think you should test drop `city=3/dt=2020-01-02` and assert `city=3/dt=2020-01-01` is existing